### DISCUSSION TITLE
Q: If I update to the latest Digital Garden Plug-in Version, which files will it overwrite?

### DISCUSSION THREAD

**Discussion by iwyatt** [oleeskild/obsidian-digital-garden#234] [Q&A] [answered]
Hi all - first of all - what a fantastic plugin! I've added it to the list of donation-supported projects I contribute to every year (usually around the holidays).

I saw in the Digital Garden repo a recent commit message that it has supported for post datestamps, which would be convenient.

### Q: If I update to the latest Digital Garden Plug-in Version, which files will it overwrite?

- My version still uses the custom css file to override subsequent CSS. Am I going to lose taht?
- I've customized some of the module container njk files - eg the sidebar file. Is that going to get overwritten?

I guess that the worst case scenario is I would need to rollback those specific files in github, but I'm, not sure the time investment to do so is worth the convenience of date stamps on pages.

Thank you!

**Comment by oleeskild** [comment, marked as answer]
Thanks for the kind words, and for adding the project to your list 😄 

The custom css file will not be overwritten. It's meant to be a place where you can put things without worrying about it being touched by the update. 
Unfortunately the sidebar file will get overwritten. (Side note: What have you customized? Perhaps you could meet your needs by using the [slot](https://dg-docs.ole.dev/advanced/adding-custom-components/#how-does-it-work) feature, which the update doesn't touch.)

That being said, clicking the update template button will only make a PR. You can then look at the changes before deciding to merge it into `main`. So there's no harm in just clicking the button. If you don't want to merge it you can just delete the PR. 

A tip for quickly modifying the code of a PR is to press `.` on your keyboard when looking at the PR in GitHub. This will open up VS Code in your browser, allowing you to modify the code before merging it in. When doing so, your PR should get updated with new links, generated by Vercel or Netlify, where you can see if the PR-build was successful or not.

**Comment by iwyatt** [comment]
Thank for the thorough info, @oleeskild ! 

re: the sidebar customization - its only ~8 lines added to sidebar.njk that hides the side bar components when the user is on the front page of the website (path = "/').:

eg in sidebar.njk:
```
 {%if settings.dgShowToc === true and page.url !== '/'%}
```

I also intend to move some of the sidebar components from the right side of the page to the left side (in the navbar), and update the 404 page with a graph and search box.

Will review the pr and update per your suggestion when I have some time. 

Thanks again!
